### DISCUSSION TITLE
Is there a tutorial of sorts for databake?

### DISCUSSION THREAD

**Discussion by madig** [unicode-org/icu4x#3021] [Q&A]
Hi!
I'm interested in trying out databake for a toy project, where I build an API around an XML data dump. My current approach is to convert the data to postcard format, include_bytes the output and deserialize on load, but having the data (essentially a vec of structs) tucked away by the compiler sounds nice. I unfortunately have no idea how to use databake and skimming the ICU4X code base does not enlighten me. Where do I start?

**Comment by Manishearth** [comment]
Have you gone through https://manishearth.github.io/blog/2022/08/03/zero-copy-3-so-zero-its-dot-dot-dot-negative/ ? Otherwise we don't really have any tutorial for this. Maybe eventually.

**Reply by sffc** [reply]
Databake by itself is very simple and bare-bones. It's designed for you to use in a script that invokes databake and then saves the result in the desired directory structure.

**Reply by madig** [reply]
Ah! Ok thanks, will have a read. Didn't make the connection to a build.rs step :)
